go/pod/pod_test.go

```diff
@@ -0,0 +1,63 @@
+package pod
+
+import (
+	"encoding/json"
+	"testing"
+
+	"github.com/iden3/go-iden3-crypto/v2/babyjub"
+)
+
+func TestCreateGoPod(t *testing.T) {
+	privKey := babyjub.PrivateKey{
+		0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 0, 1, 2, 3, 4, 5, 6, 7, 8,
+		9, 0, 1,
+	}
+	pod, err := signPod(privKey, PodEntries{
+		"A": PodValue{kind: "int", intVal: 123},
+		"B": PodValue{kind: "int", intVal: 321},
+		"G": PodValue{kind: "int", intVal: -7},
+		"D": PodValue{kind: "string", strVal: "foobar"},
+		"C": PodValue{kind: "boolean", boolVal: false},
+	})
+	if err != nil {
+		t.Fatalf("CreateGoPod failed: %v", err)
+	}
+	jsonPod, err := json.Marshal(pod)
+	if err != nil {
+		t.Fatalf("Failed to marshal pod to JSON: %v", err)
+	}
+	expectedPod := `{"entries":{"A":{"int":123},"B":{"int":321},"C":{"boolean":false},"D":{"string":"foobar"},"G":{"int":-7}},"signature":"fd75dc76f55eeb27e518ed5ebaca78a2b269e27d70cc0106b9f1e823380995ad8a2216351493ba3f50704ef3daae86b5163d6055d0c6644c4a1e64f03adc2704","signerPublicKey":"c433f7a696b7aa3a5224efb3993baf0ccd9e92eecee0c29a3f6c8208a9e81d9e"}`
```

Comment:
I notice that we're using Hex encoding for signatures and public keys here, while the JS implementation uses a non-standard base64 (with padding removed). This seems like a better choice but I have to flag up the incompatibility.

Reply:
the rust implementation also uses unpadded base64. in general, i think standardizing the serialized JSON form (also to the shorthand which eliminates unecessary "string" / "boolean" keys makes sense)